registry.ts

```diff
@@ -0,0 +1,14 @@
+const modules: Record<string, unknown> = {};
+
+export const getModule = (name: string) => {
+    return modules[name];
+}
+
+export const setModule = (name: string, component: unknown) => {
```

Comment:
I'm not a fan of this way of how we're making it so that `setModule` makes it so that whichever piece of code calls it first becomes gets to register a module. I don't think it's racey in practice, but it certainly feels that way. Should there be an expectation that the desktop app has made all of its `setModule` calls before the web app has loaded? As it is, I'm not sure if there's a way to ensure that that's the case though

Is there instead a way for us to do it so that the web app's version of a module is considered as the default, so we use the injected one if that exists and the web app's one otherwise?

Reply:
Yes, the expectation is that the Desktop App would load all of its modules before any of the web app code is loaded. This is achievable since the registry doesn't depend on anything else in the web app and won't load any other modules.

I'm up for having another way to do it, I just don't know what that is. We could set some kind of constant somewhere so that the registry knows that we should use the Desktop App module, as one idea.